app/Http/Controllers/HelplineController.php

```diff
@@ -234,8 +234,28 @@ public function dial(Request $request)
             exit();
         }
 
-        $conferences = $this->twilio->client()->conferences
-            ->read(array ("friendlyName" => $request->get('FriendlyName')));
+        // Sometime in August 2023, Twilio introduced a change in API Behavior. The conferences API
+        // now seems to be eventually consistent. Sometimes we get the conference back on the first
+        // try, and other times it takes a few tries. Retrying every half second seems to get the
+        // job done after no more than about 4 retries in the worst case. We try up to 10 times just
+        // to be safe.
+        $conferences = array();
+        for ($i = 0; $i < 10; $i++) {
+            $conferences = $this->twilio->client()
+                ->conferences
+                ->read(array("friendlyName" => $request->get('FriendlyName')));
+
+            if ($i > 0) {
```

Comment:
You could start the loop at 1, and put the log just before the sleep to remove a conditional

Reply:
The intent is to print a log line every time it retries, not skipping the last try.

If I did what you're suggesting, wouldn't the last (and successful) retry not print a log line?

Not sure what starting the loop at 1 accomplishes - all of the conference read requests are happening inside the loop, so the first try is still the first iteration.

Perhaps I'm misunderstanding what you're suggesting?

Reply:
It's pedantic, it's not really a change I'd require to approve the PR, just a preference thing. 
 
We are taking two slightly different logging approaches really. For you i represents the retries number while mine is the attempt number. Your way is logging that you made a retry and my way logs that no conferences were found in that attempt.

Mine would be more like 

"Conferences eventual consistency issue occurred in attempt 1"